webapp/src/main/webapp/src/app/groups/groups.module.ts

```diff
@@ -6,14 +6,22 @@ import { GroupService } from "./groups.service";
 import { BrowserModule } from "@angular/platform-browser";
 import { FormsModule } from "@angular/forms";
 import { DataTableModule } from "primeng/components/datatable/datatable";
+import { ModalModule } from "ngx-bootstrap";
+import { DeleteGroupModalComponent } from "./delete-group.modal";
+
 @NgModule({
   declarations: [
-    GroupsComponent
+    GroupsComponent,
+    DeleteGroupModalComponent
+  ],
+  entryComponents: [
+    DeleteGroupModalComponent
```

Comment:
This threw me off for a while when my modal wasn't working.  So modal components we need to explicitly call out in the `entryComponents` because it does not have a selector reference, is not part of a route definition, and is not part of bootstrap.  So the only way angular will know to add it to the componentFactory is if we call it out here in the `entryComponents`